Solution for problem 3

DumitruBogdan opened the pull request:
Here's why you can test my code:
In the Problem3 file, you can find the code written in a java class called "Problem3". Insert this file into a new project( i worked with Intellij), src folder. Because i also had to write UnitTest, you can find my testing functions in Problem3_UnitTest.java. To run this java class, you have to insert it in src folder( right below the Problem3.java) and run this class. Down below, near console, you will find the results, where you can see that all the tests passed.

Commit:
Solution for problem 1

Commit:
Solution for problem2

Commit:
Solution for problem3

MihaiPantiru commented:
Hey @DumitruBogdan, it would be great to add the necessary steps to run your solutions and the tests.

MihaiPantiru commented:
@DumitruBogdan This PR also contains the solution for Problem 1 and Problem 2. Can you make it to have just the solution for Problem 3?

DumitruBogdan commented:
Hello @MihaiPantiru, I will update the pull requests when I will be home.

Commit:
Problem1 and 2 deleted

DumitruBogdan commented:
Hello @MihaiPantiru , I removed the problem 2 and 3 from the branch with another commit and also updated the description, where you can find exactly how to run and test my solution.

RalucaMihaela commented on Problem3.java:
```diff
@@ -0,0 +1,18 @@
+    public static String game(Stack<Character> jenny, Stack<Character> stephanie) {
+        String result = "";
+        while (!jenny.isEmpty() || !stephanie.isEmpty()) {
+            if (jenny.isEmpty())
```
Is there a way to reduce the if-else statements? It makes the code a bit hard to read.

DumitruBogdan replied:
To reduce them, I don't think so. Because it is needed to check if the stack is empty before doing the peek() operation, which will throw an exception if the stack is empty. Still, it is possible to place them out of loop, but additional code is needed. Also, the time complexity is a good one, O(MAX(stack1.size, stack2.size). 
Indeed, it is hard to read, but they are necessary, from my point of view.

RalucaMihaela reviewed (commented):
Hy Bogdan, great solution. I left a comment for you to look over it.
